CIP-0144/README.md

```diff
@@ -0,0 +1,362 @@
+
+| API | Version |
+| --- | --- |
+| Connection-API | 0.0.0 |
+
+
+### Appendix
+
+This appendix contains additional schemas for types that are used in the APIs.
+
+#### Connection API Data Types
+
+##### Extensions
+
+```
+{
+  "type": "object",
+  "title": "Extensions",
+  "properties": {
+    "extensions": {
+      "type": "array",
+      "items": {
+          {
+            "type": "object",
+            "title": "Extension",
+            "properties": {
+              "cip": {
+                "type": "number"
+              },
+              "version: {
+                "$ref": "#/appendix/SemVer"
+              }
+            },
+            "required": ["cip"],
+          }
+      }
+    }
+  },
+  "required": ["extensions"],
+}
+```
+
+An extension is an object with a "cip" field that describes a CIP number extending the API, and an optional version specified according to SemVer. For example:
+
+```
+{ "cip": 30 }
+```
+
+
+##### SemVer
+
+```
+{
+  "type": "object",
+  "title": "SemVer",
+  "properties": {
+    "major": {
+      "title": "Major",
+      "type": "number",
+    },
+    "minor": {
+      "title": "Minor",
+      "type": "number",
+    },
+    "patch": {
+      "title": "Patch",
+      "type": "number",
+    }
+  },
+  "required": ["major", "minor", "patch"],
+}
+```
+
+#### Errors
+
+This section lists the possible errors the wallet connector may return. Each error is specified by an error code, and an optional info string describing the cause of the error. Below each error we list names for all the error codes, and a brief description of when they should be used. Other than when the spec dictates the use of a specific error code, wallets can choose the error code they deem more applicable to the situation. Wallets can also extend these errors with additional codes should they feel the need to.
+
+##### APIError
+
+```
+{
+  "type": "object",
+  "title": "APIError",
+  "properties": {
+    "code": {
+      "type": "number",
+      "title": "APIErrorCode"
+      "enum": [-1, -2, -3, -4, -5]
+    },
+    "info": {
+      "type": "string"
+    }
+  },
+  "required": ["code"],
+}
+```
+
+- `InvalidRequest`: (-1) Inputs do not conform to this spec or are otherwise invalid.
+- `InternalError`: (-2) An error occurred during execution of this API call.
+- `Refused`: (-3) The request was refused due to lack of access - e.g. wallet disconnects.
+- `AccountChange`: (-4) The account has changed. The dApp should call wallet.enable() to reestablish connection to the new account. The wallet should not ask for confirmation as the user was the one who initiated the account change in the first place.
+- `NotSatisfiable`: (-5) The request is structurally correct, but the wallet can not satisfy it for some reason.
+
```

Comment:
would a new error be needed 
for the case where a dApp requests a network at enable time
but the wallet is not connected to it?